### PR TITLE
Make PersonalSchedule Form Description Optional

### LINKED ISSUE #51: Make description on Personal Schedule page optional
## User Story: 
As a user, I can create a personal schedule with no description, so that I have more freedom to create schedules as I please

## Discussion:  
Right now, description is required on the Personal Schedule create / edit page.  But it isn't essential for any reason. It would be more user-friendly to make it optional.

### PR DESCRIPTION
**Pull request by phamaiden**
In this PR, I made the personal schedule form more user friendly by making the description field optional. Also added full test coverage.
Closes #51 

In the gif below, I demonstrate the new personal schedule form. As you can see the add form no longer requires a description and the personal schedule index page correctly displays the new schedule with no description. Similarly, the edit form doesn't require a description
Screen Recording 2024-05-29 at 6 08 33 PM

### CODE REVIEW

**Review by Frank404NotFound — DISMISSED** [review]
LGTM

**Review by ericknee — DISMISSED** [review]
LGTM

File: frontend/src/main/pages/CourseDetails/CourseDetailsIndexPage.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ import { useBackend, _useBackendMutation } from "main/utils/useBackend";
 import CourseDetailsTable from "main/components/CourseDetails/CourseDetailsTable";
 import { yyyyqToQyy } from "main/utils/quarterUtilities";
 import CourseDescriptionTable from "main/components/Courses/CourseDescriptionTable";
+import GradeHistoryTable from "main/components/GradeHistory/GradeHistoryTable";
```
Comment (pconrad) [inline frontend/src/main/pages/CourseDetails/CourseDetailsIndexPage.js:8]: Unrelated change.  Should not be in this PR

File: frontend/src/main/pages/CourseDetails/CourseDetailsIndexPage.js [hunk 2]
```diff
@@ -26,6 +27,25 @@ export default function CourseDetailsIndexPage() {
...
+    {
+      method: "GET",
+      url: `/api/gradehistory/search`,
+      params: {
+        subjectArea: subject,
+        courseNumber: course,
+      },
+    },
+  );
+
```
Comment (pconrad) [inline frontend/src/main/pages/CourseDetails/CourseDetailsIndexPage.js:48]: Unrelated code should not be in this PR

**Review by pconrad — CHANGES_REQUESTED** [review]
Please remove the unrelated code from the PR

File: frontend/src/tests/pages/PersonalSchedules/PersonalSchedulesCreatePage.test.js [hunk 3]
```diff
@@ -171,4 +171,60 @@ describe("PersonalSchedulesCreatePage tests", () => {
...
+        </MemoryRouter>
+      </QueryClientProvider>,
+    );
+
+    expect(
+      await screen.findByTestId("PersonalScheduleForm-name"),
+    ).toBeInTheDocument();
+
+    const nameField = screen.getByTestId("PersonalScheduleForm-name");
+    //const quarterField = document.querySelector("#PersonalScheduleForm-quarter");
```
Comment (pconrad) [inline frontend/src/tests/pages/PersonalSchedules/PersonalSchedulesCreatePage.test.js:199]: Commented out code ... please remove

File: frontend/src/tests/pages/PersonalSchedules/PersonalSchedulesCreatePage.test.js [hunk 4]
```diff
@@ -171,4 +171,60 @@ describe("PersonalSchedulesCreatePage tests", () => {
...
+    const quarterField = document.querySelector(
+      "#PersonalScheduleForm-quarter",
+    );
+    //const selectQuarter = getByLabelText("Quarter")
```
Comment (pconrad) [inline frontend/src/tests/pages/PersonalSchedules/PersonalSchedulesCreatePage.test.js:203]: Commented out code.  Please remove

File: frontend/src/tests/pages/PersonalSchedules/PersonalSchedulesCreatePage.test.js [hunk 5]
```diff
@@ -171,4 +171,60 @@ describe("PersonalSchedulesCreatePage tests", () => {
...
+    const submitButton = screen.getByTestId("PersonalScheduleForm-submit");
+
+    fireEvent.change(nameField, { target: { value: "SampName" } });
+    fireEvent.change(quarterField, { target: { value: "20124" } });
+    //userEvent.selectOptions(selectQuarter, "20124");
```
Comment (pconrad) [inline frontend/src/tests/pages/PersonalSchedules/PersonalSchedulesCreatePage.test.js:208]: Pls remove commented out code.

File: frontend/src/tests/pages/PersonalSchedules/PersonalSchedulesCreatePage.test.js [hunk 6]
```diff
@@ -171,4 +171,60 @@ describe("PersonalSchedulesCreatePage tests", () => {
...
+
+    expect(submitButton).toBeInTheDocument();
+
+    fireEvent.click(submitButton);
+
+    await waitFor(() => expect(axiosMock.history.post.length).toBe(1));
+
+    expect(quarterField).toHaveValue("20124");
+    //expect(setQuarter).toBeCalledWith("20124"); //need this and axiosMock below?
```
Comment (pconrad) [inline frontend/src/tests/pages/PersonalSchedules/PersonalSchedulesCreatePage.test.js:217]: Pls remove commented out code.

**Comment by pconrad** [comment]
Deployed at https://courses-qa.dokku-01.cs.ucsb.edu/

**Review by pconrad — APPROVED** [review]
LGTM.  Will deploy on courses-qa to test before merging

**Pull request merged**